plugins/version-control-systems/git/src/main/kotlin/GitRepo.kt

```diff
@@ -71,7 +71,7 @@ private data class Include(
     val name: String
 )
 
-object GitRepoCommand : CommandLineTool {
+internal object GitRepoCommand : CommandLineTool {
```

Comment:
Since when did we stop regarding any code inside plugins as public API?

Reply:
> Since when did we stop regarding any code inside plugins as public API?

It's a proposal that I made in one of the last community meetings, and this is the first time I'd like to use that argumentation to avoid excessive major version bumps in ORT.

Reply:
Hm, mind sharing the rationale why we consider public API of plugins not "public" instead of making that API "internal"?

Reply:
Because we can't, in general. Plugin classes need to be public for service loader discovery. Also a plugin's "API" is not really an API in my view that you'd use programmatically. It's just functions that happen to be public (due to technical requirements).

That's why I believe we should generally relax the semantics here, and not indicate than any change in (currently) public classes / functions of a plugin would be a breaking API change, as very likely no one is programmatically using those classes / functions anyway.

Reply:
Would it make sense then to limit the max visibility to internal for code in plugin, if it is not use by the service loader discovery?

Reply:
> Would it make sense then to limit the max visibility to internal for code in plugin, if it is not use by the service loader discovery?

Yes, and this PR is an effort to move into this direction (limited to the scope of `CommandLineTool`, as I was working on those).

> Do you think the rational would apply also to `org.ossreviewtoolkit.plugins.packageconfigurationproviders.api.PackageConfigurationProvider` which is a plugin as well?

Note that `PackageConfigurationProvider` itself is not a plugin, but the API to be implemented *by* those plugins.

But again, in general, yes, I believe plugin implementations should keep as much as possible private / internal unless required to be public by the service loader or `ort requirements` mechanisms. At least until we have a proper mechanism for a plugin to express a dependency on any other (non-core) plugin, but I don't see that being implemented any time soon.

Reply:
Thanks! I believe I got the point at least roughly now.